tests/test_dataclasses.py

```diff
@@ -1647,6 +1658,24 @@ class B(A):
     assert B(1, y=2, z=3) == B(x=1, y=2, z=3)
 
 
+def test_repr_false():
```

Comment:
You could parametrize this (using `pytest.mark.parametrize`) with a `field_constructor` variable that takes a value of either `pydantic.dataclasses.Field` or `dataclasses.field` for simplicity 👍. Then you only need one class definition, and one set of asserts.